third_party/txt/src/txt/paragraph.cc

```diff
@@ -1273,7 +1274,13 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
       left = SK_ScalarMax;
       right = SK_ScalarMin;
       for (const GlyphPosition& gp : run.positions) {
-        if (gp.code_units.start >= start && gp.code_units.end <= end) {
+        text_length_diff = (gp.code_units.end - gp.code_units.start) - 1;
+        if (text_length_diff > start) {
+          text_length_diff = start;
+        }
+        if ((gp.code_units.start >= start && gp.code_units.end <= end) ||
+            (gp.code_units.start >= (start - text_length_diff) &&
+             gp.code_units.end == end)) {
```

Comment:
In this case, it may be faster to put this check before the `gp.code_units.start >= (star...` check in order to short circuit it as this case should be comparatively rare.

On the other hand, it might remain more readable this way, so up to you :) how you would like it.